### DISCUSSION TITLE
Architecture for Rust injection on Flutter Web

### DISCUSSION THREAD

**Discussion by Desdaemon** [fzyzcjy/flutter_rust_bridge#418] [Ideas]
/cc @fzyzcjy 

I've been thinking about how to best approach #386, and one of the few problems is how we streamline the process of compiling and bundling Rust WASM code with Flutter. Presently, Flutter Web makes the process very opaque and will require HTML editing to properly inject JS code. `flutter run` on the web does not provide any methods for injecting middleware, so I'm 80% sure we will need to intercept this with our own stack, further increasing the complexity.

A fallback option would be to opt out of integrating with `flutter run` and instead defer to wasm-bindgen documentation on injecting code, so that FRB only needs to focus on generating the necessary Rust code.

**Comment by fzyzcjy** [comment]
Good question! My general idea is, the less code we write, the less buggy our code (usually) is and it is easier to maintain. (Of course, without losing performance etc). 

I am not very sure why you are going to inject the middleware? Our library does not look like a middleware.

**Reply by Desdaemon** [reply]
What I mean by middleware is the build system, for example desktop has CMake, MacOS has CocoaPods and so on. Flutter Web however is managed directly by Flutter and therefore harder to inject code in a clean manner.

**Reply by fzyzcjy** [reply]
Hmm good question. What about firstly focus on the core part, and (firstly) require the user himself to trigger the frb_codegen manually. After that is released, we can reconsider whether to integrate this into some build system, or tell user to do so. For example, I personally have some python script for Flutter build/run, so it is no problem for me.

**Comment by spelltasticsoup** [comment]
Something along these lines?

https://github.com/fzyzcjy/flutter_rust_bridge/issues/797
